### PR TITLE
Clarify development clusters for Starter vs Enterprise

### PR DESCRIPTION
**Pull request by jprcoliveira**
## When should this change go live?

- [ ] This change is not yet live and should not be merged until {ADD_DATE} (apply `hold` label or convert to draft PR)?
- [ ] There is no urgency with this change.
- [ ] This change or page is part of a marketing blog, conference talk, or something else on a schedule.
- [x] This functionality is already available but undocumented.
- [ ] This is a bug fix or security concern.

## PR Checklist

- [x] I have added changes to the relevant `/versioned_docs` directory, or they are not for an **already released version**.
- [x] I have added changes to the main `/docs` directory (aka `/next/`), or they are not for **future versions**.
- [x] My changes require an [Engineering review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned an engineering manager or tech lead as a reviewer, or my changes do not require an Engineering review.
- [x] My changes require a [technical writer review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned @christinaausley as a reviewer, or my changes do not require a technical writer review.

### CODE REVIEW

**Comment by jprcoliveira** [comment]
Waiting on @hisImminence review before backporting changes.

**Review by hisImminence — DISMISSED** [review]
Thank you!

**Comment by jprcoliveira** [comment]
@christinaausley this is now ready for your technical review. Tests are still running but I need to sign off

File: docs/components/concepts/clusters.md [hunk 1]
```diff
@@ -32,19 +32,27 @@ When your Free Trial plan expires, you are automatically transferred to the Free
 
 ## Development clusters
 
-Development clusters are recommended for development, testing, proof of concepts, and demos.
+Development clusters, available in the Starter and Enterprise plans, are recommended for development, testing, proof of concepts, and demos.
 
-Starter Plan users have one **development cluster**, with free execution for development included in their plan. Deployment and execution of models (process instances, decision instances, and task users) is provided at no cost.
+The way this type of cluster works varies depending if you are using it in the Starter or the Enterprise plan.
```
Comment (christinaausley) [inline docs/components/concepts/clusters.md:37]: @christian-konrad Do we want to capitalize "Plan" for "Enterprise Plan", for example, or leave as "Enterprise plan"? I don't have strong opinions here, but can make a styling decision if necessary.

Comment (akeller) [reply docs/components/concepts/clusters.md]: Did you mean to tag @jprcoliveira?

**Comment by akeller** [comment]
Is this PR ready to merge?

File: docs/components/concepts/clusters.md (hunk 1 above)
Comment (jprcoliveira) [reply docs/components/concepts/clusters.md]: @christinaausley feel free to make a call here to ensure we follow the correct styling here and are consistent :)

**Comment by jprcoliveira** [comment]
> Is this PR ready to merge?

@akeller / @christinaausley on my end it is (I even thought it was published already :) )

**Review by christinaausley — APPROVED** [review]

**Pull request merged**